apps/nouns-camp/src/hooks/simulation.js

```diff
@@ -0,0 +1,103 @@
+import React from "react";
+import { decodeAbiParameters, encodeFunctionData, parseAbiItem } from "viem";
+import { unparse } from "../utils/transactions";
+
+const SIMULATE_TRANSACTION_TYPES = [
+  "function-call",
+  "proxied-function-call",
+  "payable-function-call",
+  "proxied-payable-function-call",
+  "treasury-noun-transfer",
+  "escrow-noun-transfer",
+  "transfer",
+  "usdc-transfer-via-payer",
+];
+
+export const fetchSimulation = async ({
+  target,
+  value,
+  signature,
+  calldata,
+}) => {
+  if (!signature) signature = "transfer()";
```

Comment:
@werme you've been playing w/ these things for longer, is there a case where this would be wrong?

Reply:
Hmm, forgot that there was some complexity here... There could theoretically be calldata but no signature. But hey, saw that you just pass encoded data to the endpoint anyway, couldn’t you just send it as is then?

```js
const parsedTransaction = {
  to: target,
  input: calldata,
  value
};
```

Reply:
Seems to just be a concatHex https://github.com/wevm/viem/blob/a7c1ab1d101f730e8d25bbe848304c73b43564f7/src/utils/abi/encodeFunctionData.ts#L95

Reply:
Or perhaps it’s simplier to just decode and encode after all?

```js
function parse({ target, signature, calldata, value }) {
  if (signature === '')
    return {
      to: target,
      input: calldata,
      value
    }

  // Decode and encode like before
};
```

Reply:
i went with the simpler solution and seems to work. tenderly understands those calls as transfers too: https://www.tdly.co/shared/simulation/40b64036-bd66-4ba8-8a34-d50f75282060